core/src/main/java/com/datastax/oss/driver/internal/core/resolver/mockResolver/MockResolver.java

```diff
@@ -0,0 +1,83 @@
+package com.datastax.oss.driver.internal.core.resolver.mockResolver;
+
+import com.datastax.oss.driver.internal.core.resolver.Resolver;
+import com.datastax.oss.driver.internal.core.resolver.Response;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.net.UnknownHostException;
+
+/** Resolver implementation that allows returning custom results. */
+public class MockResolver implements Resolver {
+  private final MockResolverResultSource resultSource;
+
+  MockResolver(MockResolverResultSource resultSource) {
+    this.resultSource = resultSource;
+  }
+
+  /**
+   * Returns first {@link InetAddress} from predefined {@link Response} or forwards the call to
+   * {@link InetAddress#getByName(String)} if undefined.
+   *
+   * @param host host to resolve.
+   * @return address.
+   * @throws UnknownHostException
+   */
+  @Override
+  public InetAddress getByName(String host) throws UnknownHostException {
+    Response response = this.resultSource.getResponse(host);
+    if (response == null) {
+      return InetAddress.getByName(host);
+    }
+    return response.result()[0];
+  }
+
+  /**
+   * Returns all {@link InetAddress}es from predefined {@link Response} or forwards the call to
+   * {@link InetAddress#getAllByName(String)} if undefined.
+   *
+   * @param host host to resolve.
+   * @return address.
+   * @throws UnknownHostException
+   */
+  @Override
+  public InetAddress[] getAllByName(String host) throws UnknownHostException {
+    Response response = this.resultSource.getResponse(host);
+    if (response == null) {
+      return InetAddress.getAllByName(host);
+    }
+    return response.result();
+  }
+
+  /**
+   * If provided {@link SocketAddress} is unresolved instance of {@link InetSocketAddress} then the
+   * resolver will attempt to substitute underlying host and return new {@link SocketAddress} with
+   * mapped host. Otherwise or if there is no mapping defined it will return provided argument as
+   * is. In case of unsuccessful resolution of mapped hostname it will return {@link SocketAddress}
+   * with bogus hostname.
```

Comment:
```suggestion
   * If {@link addr} is an resolved {@link InetSocketAddress}, or there is no response overriding for the it's host, or {@link addr} is something else, it returns {@link addr} as is. 
   * Otherwise it returns overrided response, if overrided response throws {@link UnknownHostException}, it emulates it by returning unresolved {@link InetSocketAddress} with bogus host.
   *  
```